### ISSUE TITLE
Calibration issues with Fedora

### ISSUE THREAD

**Issue by Kovak0** [TobleMiner/wintron7.0#4]
Hello,

I'm attempting to run Fedora 27 XFCE on my Wintron 7.

I have fetched the fw file and put it in the correct folder (/lib/firmware/mssl1680.fw), now it does recognise the touch-screen but it's terribly off. It should've been already calibrated but it isn't...

What should I do now?

**Comment by TobleMiner** [comment]
Hi,

there is a tool called xinput_calibrator. I needed to calibrate my touchscreen using it since it alters the configuration of your X server. Take a look at https://www.freedesktop.org/wiki/Software/xinput_calibrator/

There is some documentation on how to use xinput_calibrator in the man page: https://linux.die.net/man/1/xinput_calibrator

**Comment by Kovak0** [comment]
Thanks for your reply.

I tried that just now, no dice, or maybe it's just me...

Would you mind sharing your own steps to do it? And which distro, so that I can at the very least figure out where to look at.

**Comment by Kovak0** [comment]
Still nothing, I can't seem to get around it.

**Comment by TobleMiner** [comment]
@Kovak0 I'll try it on the realy hardware as soon as I can. My Wintron is currently out of juice.

I'm using ArchLinux for all my experiments with cheap chineese hardware and generally work at the bleeding edge of Linux kernel development to get the latest bug fixes (aka I do quite often just clone linux master and hope for the best).

I'll add some calibration instructions to the main README as soon as my Wintron is juiced back up.

**Comment by Kovak0** [comment]
@TobleMiner

Is there any way you can add Fedora and/or Ubuntu? Both have recent kernels (Ubuntu has HWE, Fedora updates to the latest), I would love to be of any help, if I can.

TIA

**Comment by TobleMiner** [comment]
I did not really plan on doing that since this repo was always more about kernel support than anything else but I might do if I find some time for it.

I'm not sure if any of them will run well with stock kernels though. I didn't see any distros shipping with kernels that have all options I need enabled yet.

And you could certainly help by testing the distros and reporting what works/doesn't work (video, WLAN, touchscreen, etc.). I can't promise that I will find enough time to follow up on all distros since I'm involved in quite a few other projects atm but I'll try.

**Comment by Kovak0** [comment]
@TobleMiner


Of course I did, everything works as you listed (audio and BT don't, camera doesn't) on both, I was talking about calibration guides, not install etc. :)

If you could write about Arch it would be a great start, others when you have time.

**Comment by Kovak0** [comment]
@TobleMiner

Okay! I've got this! Now xinput_calibrator works, but of course I need to find the calibration matrix.

Could you share yours for portrait and landscape? Maybe add how you calculate it.

Once you're done just tell me if you want me to drop a few lines here for fedora users. :)

**Comment by TobleMiner** [comment]
The only type of calibration data I have is in ```/etc/X11/xorg.conf.d/99-calibration.conf```
```
Section "InputClass"
	Identifier	"calibration"
	MatchProduct	"Silead GSLx680 Touchscreen"
	Option	"Calibration"	"9 886 26 626"
EndSection
```
As for calculating calibration matrices there is an article on it in the Archwiki:
https://wiki.archlinux.org/index.php/Calibrating_Touchscreen#Calculate_the_Coordinate_Transformation_Matrix

**Comment by Kovak0** [comment]
@TobleMiner 

I guess I'll just wait for you to spoonfeed me and see where we can get. Could you share your steps on Arch?

**Comment by TobleMiner** [comment]
Yeah, I will. But that will probably take me some days

**Comment by Kovak0** [comment]
@TobleMiner any news?

**Comment by TobleMiner** [comment]
@Kovak0 Sorry, I was not at home for the last few days and didn't find any time today (yet) to take another look at the calibration. I'll try to figure something out this evening though

**Comment by Kovak0** [comment]
Yeah.

I must be a pain in the butt.

**Comment by Kovak0** [comment]
@TobleMiner figure anything out yet?

**Comment by TobleMiner** [comment]
Nope, my device does not switch form Text- to Videomode properly anymore. It just hangs with a console cursor on the display although the XServer does start properly. I'll try to fix it as soon as possible but I have other things I need to take care of first, sorry.

**Comment by Kovak0** [comment]
Any news on this?
